HW_3_/Task_18.py

```diff
@@ -0,0 +1,29 @@
+'''Задача 18: Требуется найти в массиве A
+самый близкий по величине элемент 
+к заданному числу X. 
+Если таких элементов несколько, вы вывести один любой. 
+Пользователь в первой строке вводит натуральное число N – количество элементов в массиве.
+В последующих  строках записаны N целых чисел Ai (можно использовать псевдогенерацию). 
+Последняя строка содержит число X
+
+*Пример:*
+
+5
+    7 -2 3 5 1
+    6
+    -> 7'''
+    
+n = int(input("Введите натуральное число n для задания количества элементов в массиве: "))
+print(n)
+import random
+lstA = [random.randint(-10, 10) for i in range(n)]
+print(lstA)
+x = int(input("Введите число Х: "))
+print(x)
+
+Diff = {}
+for i in range(len(lstA)):
+        Diff[lstA[i]] = abs((lstA[i] - x))
+print(Diff)
+keyMin = min(Diff, key = lambda k: Diff[k])
```

Comment:
Все решение этой задачи вообще может свестись напрямую к min
```suggestion
keyMin = min(lstA, key=lambda k: abs(k - x)
```